PLEP-0007.rst

```diff
@@ -0,0 +1,847 @@
+Users will add source and sink terms as necessary and choose models
+for dissipation coefficients.
+
+.. If the equations are in conservative form (including with sources and
+   sinks), then more general numerical methods may be used.
+
+
+
+Specifying the numerical method
+-------------------------------
+
+For simulations using the fluid approximation, users will choose
+between different finite difference, finite volume, finite/spectral
+element, and spectral methods. If possible, the code for the
+numerical method will be generated from the specified equations
+(though this may require that the equations be specified in
+conservative form).
+
+If the users choose a PIC simulation, then they will choose the
+particle pusher for the time advance.
+
+At this point, users will specify the numerical input parameters.
+
+.. Right now I am not certain how to deal with boundary conditions, and
+   how we could treat them similarly for fluid and PIC simulations.
+   If we have a problem setup for a fluid case that has Dirichlet BCs
+   for density, then how do we transfer that to a PIC case?  Should we
+   have BCs defined separately?  Or include them in the numerical method
+   definition maybe?  Maybe we should have a way to convert a fluid
+   simulation setup into a PIC simulation.
+
+.. Thinking more: if the boundary conditions are different between
+   fluid and PIC, then they should be denoted in different ways.  We
+   could provide some suggestions on adapting a problem for PIC vs
+   fluid approximation simulations.
+
+.. Users will next define the system of equations or physical model to be
+   solved. It is at this point that users will choose the style of
+   simulation (including but not limited to fluid, particle-in-cell, and
+   hybrid approaches). The physical model will be checked to be consistent
+   with the initial and boundary conditions.
+
+.. Users define the physical model.
+     Options for user input
+      Pre-defined sets of equations with options to specify different
+      coefficients (like resistive MHD with uniform, Spitzer, anomalous,
+      or a user-defined function)
+    List of strings containing the different equations
+   If the equations are in conservative form (including with sources
+    and sinks) then
+   We can have pre-defined sets of equations
+   We can have pre-defined sets of equations as a string (including
+   unicode characters) following Dedalus approach
+    This can be done best if numerical method can be automagically generated
+    Julia allows us to pass functions around as arguments (A function
+      can have a function as an argument, and a function can return a
+      function)
+   We can sometimes use pre-set equations (like resistive MHD, with
+     uniform or temperature dependent or anomalous resistivity)
+     Numerical method
+     Post-processing
+     Maybe we could create a function that automatically writes text that
+       describes the numerical method and such.
```

Comment:
It sounds like this could use heavy collaboration with other packages in the (completely unknown to me) Julia ecosystem... I'm thinking SciPy for PDEs.

Heck, maybe that's what we should have been trying to do instead of/alongside PlasmaPy all along.